editor.planx.uk/src/pages/FlowEditor/components/Sidebar/PublishDialog.tsx

```diff
@@ -0,0 +1,266 @@
+        <Box pb={2}>
+          <List sx={{ listStyleType: "disc", marginLeft: 3 }}>
+            <ListItem key={"updated"} disablePadding sx={{ display: "list-item" }}>
+              <Typography variant="body2">{`${changeSummary["updated"]} nodes have been updated or added`}</Typography>
+            </ListItem>
+            <ListItem key={"deleted"} disablePadding sx={{ display: "list-item" }}>
+              <Typography variant="body2">{`${changeSummary["deleted"]} nodes have been deleted`}</Typography>
+            </ListItem>
+          </List>
+          <Box
+            style={{
+              display: "flex",
+              flexDirection: "column",
+              marginBottom: 2,
+            }}
+          >
+            <Box
+              style={{
+                display: "flex",
+                flexDirection: "row",
+                alignItems: "center",
+              }}
+            >
+              <Typography variant="body2">{`See detailed changelog `}</Typography>
+              <Button
+                onClick={() => setExpandNodes((expandNodes) => !expandNodes)}
+                size="small"
+                disableRipple
+              >
+                <Caret
+                  expanded={expandNodes}
+                  color="primary"
+                  titleAccess={
+                    expandNodes ? "Less information" : "More information"
+                  }
+                />
+              </Button>
+            </Box>
+            <Collapse in={expandNodes}>
+              <Box pb={1}>
+                <ul>
+                  {alteredNodes.map((node) => (
+                    <AlteredNodeListItem node={node} />
+                  ))}
+                </ul>
+              </Box>
+            </Collapse>
+          </Box>
+        </Box>
+      )}
+      <Divider />
+      {changeSummary["portals"].length > 0 && (
+        <>
+          <Box pt={2}>
+            <Typography variant="body2">{`This includes recently published changes in the following nested services:`}</Typography>
+            <List sx={{ listStyleType: "disc", marginLeft: 3 }}>
+              {changeSummary["portals"].map((portal) => (
+                <AlteredNestedFlowListItem {...portal} />
+              ))}
+            </List>
+          </Box>
+          <Divider />
+        </>
+      )}
+    </Box>
+  );
+};
+
+export interface ValidationCheck {
+  title: string;
+  status: "Pass" | "Fail" | "Not applicable";
+  message: string;
+}
+
+export const ValidationChecks = (props: {
+  validationChecks: ValidationCheck[]
+}) => {
+  const { validationChecks } = props;
+
+  const getIcon = (status: ValidationCheck["status"]) => {
+    if (status === "Pass") {
+      return <Done color="success" />;
+    } else if (status === "Fail") {
+      return <Close color="error" />;
+    } else if (status === "Not applicable") {
+      return <NotInterested color="disabled" />;
+    }
+  }
```

Comment:
nit: An object here might be slightly more readable and typesafe, e.g. 

```ts
const Icon: Record<ValidationCheck["status"], React.FC> = {
  Pass: <Done color="success" />,
  Fail: <Close color="error" />,
  "Not applicable": <NotInterested color="disabled" />
}
```

Reply:
Yes, nice suggestion - updated!